### PR TITLE
dev: add type annotations to view

### PR DESCRIPTION
**Pull request by maratori**

### CODE REVIEW

File: lona/server.py [hunk 1]
```diff
@@ -32,6 +40,8 @@
 http_logger = logging.getLogger('lona.server.http')
 websockets_logger = logging.getLogger('lona.server.websockets')
 
+_T = TypeVar('_T')
```
Comment (fscherf) [inline lona/server.py:43]: Can we give this a better name?

File: lona/view.py [hunk 2]
```diff
@@ -1,29 +1,90 @@
...
+from lona.request import Request
+
+# avoid import cycles
+if TYPE_CHECKING:
+    from lona.server import LonaServer
+
+
+_T = TypeVar('_T')
+_V = TypeVar('_V', bound='LonaView')
+_HTML = Union[None, AbstractNode, str]
```
Comment (fscherf) [inline lona/view.py:44]: Also here. Does it make sense to collect such types in a dedicated module to make the reusable?

File: lona/view.py [hunk 3]
```diff
@@ -1,29 +1,90 @@
-from typing import List, Dict
+from typing import (
```
[the remark this answers is not in the document]
Comment (maratori) [reply lona/view.py:1]: Yep

File: lona/server.py [hunk 4]
```diff
@@ -1,10 +1,18 @@
...
+from concurrent.futures import Future
 from functools import reduce
 import operator
 import logging
 import inspect
 import asyncio
+import sys
 import os
 
+if sys.version_info >= (3, 8):
```
[the remark this answers is not in the document]
Comment (maratori) [reply lona/server.py:11]: Let's drop `if` and just import unconditionally
```py
from typing_extensions import Literal
```

File: lona/server.py (hunk 1 above)
Comment (maratori) [reply lona/server.py:43]: What about removing underscore?

Comment (fscherf) [reply lona/server.py:43]: Aren't one character definitions an anti-pattern?

Comment (maratori) [reply lona/server.py:43]: In this case no. That's how mypy recommends define generics.
https://mypy.readthedocs.io/en/stable/generics.html

Comment (maratori) [reply lona/server.py:43]: And python docs as well https://docs.python.org/3/library/typing.html#generics

Comment (fscherf) [reply lona/server.py:43]: @maratori Ah ok. I am ok with that then

File: lona/server.py [hunk 5]
```diff
@@ -1,10 +1,18 @@
...
+    from typing import Literal
+else:
+    from typing_extensions import Literal
```
Comment (fscherf) [inline lona/server.py:14]: @maratori in the last pr we agreed on just using this import in all python versions. I would do the same here

Comment (maratori) [reply lona/server.py]: It was current pr :)
I pushed the changes just now.

Comment (fscherf) [reply lona/server.py]: @maratori: Whoops :D thanks!

File: lona/view.py (hunk 2 above)
Comment (maratori) [reply lona/view.py:39]: I don't think it's a good idea. None of them truly reusable.
`_T` is just any type.
`_V` is a user view and seems is used only in this file.
`_HTML` is also used only here. Different type `Union[AbstractNode, str]` will be used in other files.

File: lona/server.py (hunk 5 above)
Comment (maratori) [reply lona/server.py]: np :)

Comment (maratori) [reply lona/server.py]: btw: is it necessary to mention you in each comment with @fscherf ?

**Comment by fscherf** [comment]
@maratori the commit message could use some more documentation but in general i am ok with the changes. I have to research some typing things more closely tomorrow (for example i diddn't know ``typing.cast`` before) but everything seems reasonable and clean. Good work! :)

**Comment by maratori** [comment]
> the commit message could use some more documentation

@fscherf What documentation do you mean?

>  I have to research some typing things more closely tomorrow

Sure, take your time.

**Comment by fscherf** [comment]
@maratori I thought it would make sense to document the changed import for ``typing_extentsions`` but i think it's clear enough. I'll take it like this :) Thanks!

**Pull request merged**